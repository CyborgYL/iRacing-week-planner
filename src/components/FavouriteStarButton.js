// @flow

import * as React from 'react';
import classnames from 'classnames';

import styles from '../styles/main.module.scss';

type Props = {
  onClick?: (newValue: boolean) => void,
  enabled?: boolean,
};

export default function FavouriteStarButton({ enabled, onClick }: Props): React.Node {
  return (
    <span
      onClick={() => onClick(!enabled)}
      className={classnames({
        [styles.glyphicon]: true,
        [styles['glyphicon-star']]: enabled,
        [styles['glyphicon-star-empty']]: !enabled
      })}
    />
  );
}

FavouriteStarButton.defaultProps = {
  enabled: false,
  onClick: () => {},
};
