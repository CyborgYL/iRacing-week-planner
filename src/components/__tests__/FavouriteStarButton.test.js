import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';

import FavouriteStarButton from '../FavouriteStarButton';

describe('components/FavouriteStarButton', () => {
  test('renders correctly', () => {
    const component = shallow(<FavouriteStarButton />);

    component.simulate('click');

    expect(component).toMatchSnapshot();
  });

  test('renders enabled with onClick', () => {
    const onClick = jest.fn();
    const component = shallow(<FavouriteStarButton enabled onClick={onClick} />);

    component.simulate('click');

    expect(component).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledWith(false);
  });

  test('renders disabled with onClick', () => {
    const onClick = jest.fn();
    const component = shallow(<FavouriteStarButton onClick={onClick} />);

    component.simulate('click');

    expect(component).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledWith(true);
  });
});
