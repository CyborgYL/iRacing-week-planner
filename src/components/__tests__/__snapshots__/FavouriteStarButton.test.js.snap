// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FavouriteStarButton renders correctly 1`] = `
<span
  className="glyphicon glyphicon-star-empty"
  onClick={[Function]}
/>
`;

exports[`components/FavouriteStarButton renders disabled with onClick 1`] = `
<span
  className="glyphicon glyphicon-star-empty"
  onClick={[Function]}
/>
`;

exports[`components/FavouriteStarButton renders enabled with onClick 1`] = `
<span
  className="glyphicon glyphicon-star"
  onClick={[Function]}
/>
`;
